packaging/hip-targets.cmake

```diff
@@ -39,22 +39,25 @@ unset(_expectedTargets)
 
 
 # The installation prefix configured by this project.
-set(_IMPORT_PREFIX "/opt/rocm/hip")
-
+if( DEFINED ENV{ROCM_PATH} )
+     set(_IMPORT_PREFIX "$ENV{ROCM_PATH}/hip")
+else()
+     set(_IMPORT_PREFIX "/opt/rocm/hip")
+endif()
 # Create imported target hip::hip_hcc_static
 add_library(hip::hip_hcc_static STATIC IMPORTED)
 
 set_target_properties(hip::hip_hcc_static PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;/opt/rocm/hsa/include"
-  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;/opt/rocm/hsa/include"
+  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;$ENV{ROCM_PATH}/hsa/include"
```

Comment:
The hip-config.cmake should search for hsa and use the path found by cmake. You can see how hcc does it [here](https://github.com/RadeonOpenCompute/hcc/blob/clang_tot_upgrade/lib/hcc-config.cmake.in#L7).

Reply:
@pfultz2 The issue is that our build system (rocm-ci) does not have hsa and hcc installed in the same paths as the target system. So paths have to be determined based on target install locations.
@paulfreddy Again we need consistency in ROCM_PATH, HSA_PATH, HCC_HOME evaluation.

Reply:
> The issue is that our build system (rocm-ci) does not have hsa and hcc installed in the same paths as the target system

Yes, exactly, and its not just rocm-ci that does this, package managers like spack or cget can install everything in different directories. 

This why it needs to call `find_*` commands to find the dependencies. Hip already does this for hcc by calling `find_dependency(hcc)` [here](https://github.com/ROCm-Developer-Tools/HIP/blob/master/hip-config-hcc.cmake.in#L53). However, it missing a find command for hsa runtime headers. 

Hip needs to call `find_path(HSA_HEADER hsa/hsa.h)` to get the headers or it could rely on hcc to already find hsa and then just use either the `HSA_HEADER` variable or use the `hsa-runtime64` imported target. 

Hcc does it properly and it doesnt require relying on custom env variables or using incorrect relative paths. Hcc also works with rocm-ci and other package managers like cget.

Hip should do this properly as well at least for cmake(hipcc is just a perl script that we are not compelled to use so its not urgent to fix).